File: 1-TwoSum/1-TwoSum.java
// Last updated: 7/25/2025, 8:55:19 AM
class Solution {
    public int[] twoSum(int[] arr, int target) {

        HashMap<Integer, Integer> mpp = new HashMap<>();

        for(int i =0; i<arr.length; i++){

            int k = target- arr[i];
            if(mpp.containsKey(k)){
                return new int[]{mpp.get(k), i};
            }

            mpp.put(arr[i], i);
        }

        return new int[]{};
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 7/31/2025, 8:12:53 PM


class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map; 

        for (int i = 0; i < nums.size(); ++i) {
            int complement = target - nums[i];

            if (map.count(complement)) {
                return {map[complement], i};
            }

            map[nums[i]] = i;
        }

        return {}; 
    }
};
